ThunderCloud/NavigationController.swift

```diff
@@ -665,6 +665,36 @@ public extension UINavigationController {
         }
     }
     
+    /// # Context
+    /// If a `UIViewController` is being pushed on the navigation controller stack and we
+    /// do not want the tab bar at the bottom of the screeen we set
+    /// `hidesBottomBarWhenPushed = true`.
+    ///
+    /// # Issue
+    /// Given a `UINavigationController`, in a `UITabBarController`,  with a
+    /// viewController stack like:
+    /// - `rootViewController`
+    /// - `viewController1`
+    /// - `viewController2`
+    /// where want the tab bar hidden for both `viewController1` and `viewController2` but not
+    /// on the `rootViewController`.
+    /// Then the correct implementation would be to set `hidesBottomBarWhenPushed = true`
+    /// on and before pushing `viewController1`. We don't need to set it also
+    /// for `viewController2`.
+    ///
+    /// If we also set `hidesBottomBarWhenPushed = true` for `viewController2` then
+    /// calling `popToRootViewController` from `viewController2` would pop back
+    /// to `rootViewController` but the tab bar would still be hidden.
+    ///
+    /// # Solution
+    /// Before pushing a `UIViewController` which wants the tab bar hidden, set
+    /// `hidesBottomBarWhenPushed` to the `return` of this.
+    /// This function checks to see if there is another `UIViewController` in the navigation
+    /// controller stack (before the to be pushed) which has `hidesBottomBarWhenPushed = true`
+    func shouldHideBottomBarWhenPushed() -> Bool {
```

Comment:
Dem docs tho! 👏🏻

Reply:
🙇  Looking for those "Exceeding exceptions" 😂

Reply:
Haha so good!